### ISSUE TITLE
Add button "Sort A-Z" in interface to manage favorites

### ISSUE THREAD

**Issue by falzonv** [falzonv/discreet-launcher#268]
Received by mail:

> Thank u for the great app!
Wish it had option to arrange favorites by alphabetical order without having to keep moving manually.
Thnx again

**Answer:**

Thank you for your message, I am glad you like Discreet Launcher!

Regarding your suggestion, could you please clarify if you mean:
~~1/ A setting to automatically keep favorites alphabetically ordered when some are added/removed? (in that case the interface to arrange favorites would be hidden when the setting is enabled to avoid conflicts)~~
2/ A button such as "Sort A -> Z" in the interface to manage favorites that would replace the current order by an alphabetical order? (button triggered manually to avoid conflicts)

**Comment by falzonv** [comment]
**User's answer:**
> The button to sort a-z would be great!
> Thnx

**My answer:**
Ok, I will have a look into Sort A-Z then!
(Just so that you are not surprised, it will take some time before being implemented because I am currently very busy with my job.)

**Comment by itsmedaniel96** [comment]
Agreed! If this launcher can somehow sort apps by alphabet, or by genre; like music, cameras, pic editors, browsers. Then, i would definitely installl this app!

**Comment by falzonv** [comment]
Hello @itsmedaniel96,

Actually the main app list is always ordered alphabetically while the favorites can be ordered manually. The purpose of this feature request is to add a button allowing to re-order the favorites by A-Z rather than doing it manually.

Sorting by genre will not be implemented as this not something I am able to do (there is no reliable way to detect the genre of all installed apps programmatically). I am also not sure it would be really relevant: most people probably have only 1 or 2 apps of the same genre so the end result may just look like an unordered list.

Best regards.

**Comment by jcperil** [comment]
I prefer the manual approach because it gives me the freedom to place my favorites as I like. I hope it stays as an alternative option to alphabetical sorting.

**Comment by falzonv** [comment]
@jcperil:
Thank you for your feedback!
From the two options in the initial post, only the second one will be implemented (I just stroked the first option for clarity) so it will not change the ability to manually arrange favorites. The new button "Sort A-Z" will simply allow to re-order them alphabetically, as if the user did it manually one by one (not sure yet where this button will be displayed, probably next to "Select favorites").

Your comment makes me think that it would probably be a good idea to ask for confirmation before performing the A-Z re-ordering, some users have lots of favorites and a click by mistake could lead to unpleasant results.

**Comment by jcperil** [comment]
> Your comment makes me think that it would probably be a good idea to ask for confirmation before performing the A-Z re-ordering, some users have lots of favorites and a click by mistake could lead to unpleasant results.

Right, that would alert the users briefly.
In case, someone misclicks A-Z re-ordering, there may be a way to recover the manual ordering: When you export app data, the favorites appear in your preferred order in the `.txt` file. So, while asking for confirmation, perhaps users can also be alerted to export app data before re-ordering A-Z.

**Comment by falzonv** [comment]
Hello,

The button to sort alphabetically has been added in new release v7.2.0!
I will leave this ticket open until it is available on F-Droid (usually takes 4 to 8 days).

@jcperil: I just made a confirmation dialog so that the user has an opportunity to cancel before the change is applied. I don't want to make it too complicated either, for users who may really want to reorder favorites alphabetically.
(In case a user wrongly clicks the reordering button then wrongly confirms when asked, that's unfortunate. But the worst that can happen is that he will have to reorder his favorites, which is probably acceptable under these circumstances.)

Best regards.
